TUM Campus App/LoginViewController.swift

```diff
@@ -31,81 +30,71 @@ class LoginViewController: UIViewController {
     var logoView: TUMLogoView?
     var manager: TumDataManager?
     
-    @IBAction func skip() {
+    @IBAction func continueWithoutTumId() {
         view.endEditing(true)
         dismiss(animated: true, completion: nil)
     }
 
-    @IBAction func confirm() {
-        PersistentUser.value = .requestingToken(lrzID: getLRZ())
+    private func openTokenAuthorizationView(tumId: String) {
+        PersistentUser.value = .requestingToken(lrzID: tumId)
         performSegue(withIdentifier: "waitForConfirmation", sender: self)
     }
     
-    @IBAction func textFieldEditingChanged(_ sender: UITextField) {
-        if sender === firstTextField {
-            handleTextFieldInput(currentTextField: firstTextField,
-                                 nextTextField: numbersTextField,
-                                 characterLimit: 2)
-        } else if sender === numbersTextField {
-            handleTextFieldInput(currentTextField: numbersTextField,
-                                 previousTextField: firstTextField,
-                                 nextTextField: secondTextField,
-                                 characterLimit: 2)
-        } else if sender === secondTextField {
-            handleTextFieldInput(currentTextField: secondTextField,
-                                 previousTextField: numbersTextField,
-                                 characterLimit: 3)
-        }
-
-        confirmButton.isEnabled = textFieldContentsAreValid()
-        confirmButton.alpha = textFieldContentsAreValid() ? 1 : 0.5
+    @IBAction func openInputDialog() {
+        let alert = UIAlertController(title: "Login", message: "Enter your TUM ID", preferredStyle: .alert)
+        alert.addTextField(configurationHandler: nil) // TODO
+        alert.addAction(
+            UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        )
+        alert.addAction(
+            UIAlertAction(title: "Continue", style: .default) { action in
+                guard let text = alert.textFields![0].text else {
```

Comment:
Do you need to force unwrap here?

Reply:
Good catch 👍 I changed it to a chain of optionals.